test/config.spec.ts

```diff
@@ -0,0 +1,146 @@
+    const HOST = '0.0.0.0'
+    const PORT = '5000'
+
+    const config = getServerConfig({ HOST, PORT })
+    expect(config.host).toEqual(HOST)
+    expect(config.port).toEqual(Number(PORT))
+  })
+})
+
+describe('getStampsConfig', () => {
+  const POSTAGE_STAMP = 'f1e4ff753ea1cb923269ed0cda909d13a10d624719edf261e196584e9e764e50'
+  const POSTAGE_AMOUNT = '100'
+  const POSTAGE_DEPTH = '20'
+  const BEE_DEBUG_API_URL = 'http://localhost:1635'
+  const POSTAGE_USAGE_THRESHOLD = '0.6'
+  const POSTAGE_USAGE_MAX = '0.8'
+  const POSTAGE_TTL_MIN = '200'
+  const POSTAGE_REFRESH_PERIOD = '10'
+
+  const values: { env: EnvironmentVariables; output: StampsConfig | undefined }[] = [
+    // Postage stamps are disabld
+    { env: {}, output: undefined },
+    // Hardcoded postage stamp
+    { env: { POSTAGE_STAMP }, output: { mode: 'hardcoded', stamp: POSTAGE_STAMP } },
+    // Shoudl return hardcoded postage stamp even if autobuy values are provided
+    {
+      env: { POSTAGE_STAMP, POSTAGE_AMOUNT, POSTAGE_DEPTH, BEE_DEBUG_API_URL },
+      output: { mode: 'hardcoded', stamp: POSTAGE_STAMP },
+    },
+    // Autobuy with default values
+    {
+      env: { POSTAGE_AMOUNT, POSTAGE_DEPTH, BEE_DEBUG_API_URL },
+      output: {
+        mode: 'autobuy',
+        depth: Number(POSTAGE_DEPTH),
+        amount: POSTAGE_AMOUNT,
+        beeDebugApiUrl: BEE_DEBUG_API_URL,
+        usageTreshold: DEFAULT_POSTAGE_USAGE_THRESHOLD,
+        usageMax: DEFAULT_POSTAGE_USAGE_MAX,
+        ttlMin: (DEFAULT_POSTAGE_REFRESH_PERIOD / 1000) * 5,
+        refreshPeriod: DEFAULT_POSTAGE_REFRESH_PERIOD,
+      },
+    },
+    // Autobuy with all values set
+    {
+      env: {
+        POSTAGE_AMOUNT,
+        POSTAGE_DEPTH,
+        BEE_DEBUG_API_URL,
+        POSTAGE_USAGE_MAX,
+        POSTAGE_USAGE_THRESHOLD,
+        POSTAGE_TTL_MIN,
+        POSTAGE_REFRESH_PERIOD,
+      },
+      output: {
+        mode: 'autobuy',
+        depth: Number(POSTAGE_DEPTH),
+        amount: POSTAGE_AMOUNT,
+        beeDebugApiUrl: BEE_DEBUG_API_URL,
+        usageTreshold: Number(POSTAGE_USAGE_THRESHOLD),
+        usageMax: Number(POSTAGE_USAGE_MAX),
+        ttlMin: Number(POSTAGE_TTL_MIN),
+        refreshPeriod: Number(POSTAGE_REFRESH_PERIOD),
+      },
+    },
+    {
+      // Autobuy TTL should be 5 times refresh period if not provided
+      env: { POSTAGE_AMOUNT, POSTAGE_DEPTH, BEE_DEBUG_API_URL, POSTAGE_REFRESH_PERIOD },
+      output: {
+        mode: 'autobuy',
+        depth: Number(POSTAGE_DEPTH),
+        amount: POSTAGE_AMOUNT,
+        beeDebugApiUrl: BEE_DEBUG_API_URL,
+        usageTreshold: DEFAULT_POSTAGE_USAGE_THRESHOLD,
+        usageMax: DEFAULT_POSTAGE_USAGE_MAX,
+        ttlMin: (Number(POSTAGE_REFRESH_PERIOD) / 1000) * 5,
+        refreshPeriod: Number(POSTAGE_REFRESH_PERIOD),
+      },
+    },
+  ]
+
+  values.forEach(({ env, output }) => {
+    it(`should return ${JSON.stringify(output)} for ${JSON.stringify(env)}`, () => {
```

Comment:
Ufff this is rather crazy to put `JSON.stringify` into name of test 🥸

Maybe you could introduce property `name` for the object where you would put pretty much the content of the comments that you have already for the test cases and use that for the name of tests?